containers/src/Kokkos_DynRankView.hpp

```diff
@@ -599,12 +599,11 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   // This assumes a contiguous underlying memory (i.e. no padding, no
   // striding... AND a Trilinos/Sacado scalar type )
   template <typename iType>
-  KOKKOS_INLINE_FUNCTION std::enable_if_t<
-      std::is_integral_v<iType> &&
-          !std::is_same_v<typename drvtraits::value_type,
-                          typename drvtraits::scalar_array_type>,
-      reference_type>
-  operator[](const iType& i0) const {
+  KOKKOS_INLINE_FUNCTION
+      std::enable_if_t<!std::is_same_v<typename drvtraits::value_type,
+                                       typename drvtraits::scalar_array_type>,
+                       reference_type>
+      operator[](const iType& i0) const {
```

Comment:
The SFINAE doesn't depend on a template argument anymore.

Reply:
Please fix it and push to my branch

Reply:
Clearly I didn't look carefully enough